File: 0070-climbing-stairs/0070-climbing-stairs.py
class Solution:
    def climbStairs(self, n: int) -> int:
        if n <= 3:
            return n
        ways = [0] * (n+1)
        ways[1] = 1
        ways[2] = 2
        ways[3] = 3
        for i in range(4, n+1):
            ways[i] = ways[i-1] + ways[i-2]
        return ways[n] 

File: 0070-climbing-stairs/0070-climbing-stairs.cpp
class Solution {
public:
    int climbStairs(int n) {
        //if (n == 1) return 1;
        //if (n == 2) return 2;
        //return climbStairs(n-1) + climbStairs(n-2); 
        vector<int> dp;
        dp.push_back(0);
        dp.push_back(1);
        dp.push_back(2);
        for (int i = 3; i <= n; i++) {
            dp.push_back(dp[i-1]+dp[i-2]);
        }
        return dp[n];
    }
};